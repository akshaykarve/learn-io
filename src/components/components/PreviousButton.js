import React from 'react';
import '../ComponentStyle.css';

const PreviousButton=(props)=>{
	let but;
	if(props.skip===0){
		but=<button disabled style={{color:'grey'}} className='homeButton'>Previous</button>
	}else{
		but=<button className='homeButton' onClick={()=>{props.setSkip(props.skip-props.limit)}}>Previous</button>
	}
	return but;
}

export default PreviousButton;