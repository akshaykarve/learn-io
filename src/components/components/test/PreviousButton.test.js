import React from 'react';
import renderer from 'react-test-renderer';
import {default as PreviousButton} from "../PreviousButton";

describe("Previous Button Component", ()=>{
	it('first page', () => {
		
		const tree = renderer.create(
			<PreviousButton skip={0}
			/>
		).toJSON();
		expect(tree).toMatchSnapshot();
	});
	it('skip some platforms to show next page', () => {
		
		const tree = renderer.create(
			<PreviousButton skip={10}
			/>
		).toJSON();
		expect(tree).toMatchSnapshot();
	});
});