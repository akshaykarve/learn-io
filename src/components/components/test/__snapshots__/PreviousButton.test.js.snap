// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Previous Button Component first page 1`] = `
<button
  className="homeButton"
  disabled={true}
  style={
    Object {
      "color": "grey",
    }
  }
>
  Previous
</button>
`;

exports[`Previous Button Component skip some platforms to show next page 1`] = `
<button
  className="homeButton"
  onClick={[Function]}
>
  Previous
</button>
`;
